### ISSUE TITLE
Cross compile Windows release on Travis CI

### ISSUE THREAD

**Issue by PeterDaveHello** [leonsoft-kras/transmisson-remote-gui#991]
It'll be really really helpful if we can build the Windows release on Travis CI.

**Comment by leonsoft-kras** [comment]
I don't know. It's not hard for me to make a distribution once a month.
What's the problem? what's the question?

**Comment by PeterDaveHello** [comment]
It's not hard for me at all, the question should be: if we can do it better, why **not**? Continuous integration and continuous delivery is the trend, help assurance software quality, also, we can easily test each commit as each build, so why not? It's not a **must** but why should we against being better?

**Comment by leonsoft-kras** [comment]
I thought you had a problem.

Tell me, what about the installer (inno)? To create a distribution, you need windows anyway.

**Comment by PeterDaveHello** [comment]
Didn't you say that we will dropped it as I already provided zip release? I'm confused 😕

**Comment by leonsoft-kras** [comment]
No, I never said to remove the distribution. Me and many users don't need a zip archive.
Most people don't want to go around unpacking and replacing files in a folder.

**Comment by PeterDaveHello** [comment]
Okay, maybe I misunderstood, mind to tell me more detail about this:

> I understand the compilation for Windows is no longer needed?

https://github.com/leonsoft-kras/transmisson-remote-gui/issues/973#issuecomment-317239524

At least, we can still provide both, and maybe I can even build the installer on Linux, it won't hurt anyway, I opened this issue is just because I'm not sure if I have enough time to figure the process out, if somebody wants to pick and help, why not? It's not force anyone to use a range of time already arranged for something else, just free time for open source contribution.

**Comment by leonsoft-kras** [comment]
1. There are no obstacles to make a version in the Travis.
2. There are no obstacles to archive version of the Travis for 0.00001% of people who can not live without portable version.
3. The version for Windows with the installer should be done every time.

**Comment by antekgla** [comment]
http://wiki.lazarus.freepascal.org/Cross_compiling_for_Win32_under_Linux

Maybe this can help. I dont know ANYTHING about Linux so is not my thing.

**Comment by PeterDaveHello** [comment]
I tested it a little bit before but didn't figure it out, haven't learned lararus / pascal yet lol.

**Comment by PeterDaveHello** [comment]
Resolved by #1564. Windows x86 and x86_64 installers and portable ZIP packages are now built and validated on native Windows GitHub Actions runners for pull requests and pushes to `master`.

The workflow exercises the existing `make_setup.bat` and `make_zipdist.bat` release entry points, validates architecture and version metadata, checks UPX integrity for portable packages, smoke-tests silent installation and removal, and uploads the resulting artifacts. This supersedes the original Travis CI cross-compilation proposal.
